doc/user/methods/random_forest_regressor.md

```diff
@@ -0,0 +1,639 @@
+{
+ public:
+  // If a split with better resulting gain than `bestGain` is found, then
+  // information about the new, better split should be stored in `splitInfo` and
+  // `aux`.  Specifically, a split is better than `bestGain` if the sum of the
+  // gains that the children will have (call this `sumChildrenGains`) is
+  // sufficiently better than the gain of the unsplit node (call this
+  // `unsplitGain`):
+  //
+  //    split if `sumChildrenGains - unsplitGain > bestGain`, and
+  //             `sumChildrenGains - unsplitGain > minGainSplit`, and
+  //             each child will have at least `minLeafSize` points
+  //
+  // The new best split value should be returned (or anything greater than or
+  // equal to `bestGain` if no better split is found).
+  //
+  // If a new best split is found, then `splitInfo` and `aux` should be
+  // populated with the information that will be needed for
+  // `CalculateDirection()` to successfully choose the child for a given point.
+  // `splitInfo` should be set to a vector of length 1.  The format of `aux` is
+  // arbitrary and is detailed more below.
+  //
+  // If `UseWeights` is false, the vector `weights` should be ignored.
+  // Otherwise, they are instance weighs for each value in `data` (one
+  // categorical dimension of the input data, which takes values between `0` and
+  // `numCategories - 1`).
+  template<bool UseWeights, typename VecType, typename ResponsesType,
+           typename WeightVecType>
+  static double SplitIfBetter(
+      const double bestGain,
+      const VecType& data,
+      const size_t numCategories,
+      const ResponsesType& labels,
+      const WeightVecType& weights,
+      const size_t minLeafSize,
+      const double minGainSplit,
+      arma::vec& splitInfo,
+      AuxiliarySplitInfo& aux,
+      FitnessFunction& fitnessFunction);
+
+  // Return the number of children for a given split (stored as the single
+  // element from `splitInfo` and auxiliary data `aux` in `SplitIfBetter()`).
+  size_t NumChildren(const double& splitInfo,
+                     const AuxiliarySplitInfo& aux);
+
+  // Given a point with (categorical) value `point`, and split information
+  // `splitInfo` and `aux`, return the index of the child that corresponds to
+  // the point.  So, e.g., for `AllCategoricalSplit`, which splits a categorical
+  // dimension into one child for each category, this simply returns `point`.
+  template<typename ElemType>
+  static size_t CalculateDirection(
+      const ElemType& point,
+      const double& splitInfo,
+      const AuxiliarySplitInfo& /* aux */);
+
+  // This class can hold any extra data that is necessary to encode a split.  It
+  // should only be non-empty if a single `double` value cannot be used to hold
+  // the information corresponding to a split.
+  class AuxiliarySplitInfo { };
+};
+```
+
+***Note:*** this API differs from the `CategoricalSplitType` API required for
+[`DecisionTree`](decision_tree.md)! <!-- TODO: fix link! -->
```

Comment:
```suggestion
[`DecisionTree`](decision_tree.md#categoricalsplittype)!
```
I think this will work.

There are a couple comments I have left along these lines; another one that occurs to me now is that it might make sense to use `RandomForest` instead of `DecisionTree` as the "other" API we are referencing.